pghoard/restore.py

```diff
@@ -788,8 +807,14 @@ def _write_status_output_to_file(self, output_file):
             }
         )
 
-    def _wait_for_jobs_to_complete(self):
+    def _wait_for_jobs_to_complete(self, pool):
         while self.jobs_in_progress():
+            with self.lock:
+                for item in self.data_files:
```

Comment:
This could check that `jobs_to_retry` is not empty so that it doesn't needlessly iterate over all data files as there almost never is anything to retry.

Reply:
Done